cmd/install/assets/hypershift_operator.go

```diff
@@ -103,6 +104,141 @@ func (o HyperShiftOperatorOIDCProviderS3Secret) Build() *corev1.Secret {
+}
+
+func (o ExternalDNSDeployment) Build() *appsv1.Deployment {
+	replicas := int32(1)
+	deployment := &appsv1.Deployment{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Deployment",
+			APIVersion: appsv1.SchemeGroupVersion.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "external-dns",
+			Namespace: o.Namespace.Name,
+		},
+		Spec: appsv1.DeploymentSpec{
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"name": "external-dns",
+				},
+			},
+			Replicas: &replicas,
+			Template: corev1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"name":                    "external-dns",
+						"app":                     "external-dns",
+						hyperv1.OperatorComponent: "external-dns",
+					},
+				},
+				Spec: corev1.PodSpec{
+					ServiceAccountName: o.ServiceAccount.Name,
+					Containers: []corev1.Container{
+						{
+							Name:            "external-dns",
+							Image:           o.Image,
+							ImagePullPolicy: corev1.PullIfNotPresent,
+							Command:         []string{"/external-dns"},
+							Args: []string{
+								"--source=service",
+								"--source=openshift-route",
+								fmt.Sprintf("--domain-filter=%s", o.DomainFilter),
+								fmt.Sprintf("--provider=%s", o.Provider),
+								"--aws-zone-type=public",
+								"--registry=noop",
+								"--txt-owner-id=hypershift",
+							},
+							Ports: []corev1.ContainerPort{{Name: "metrics", ContainerPort: 7979}},
+							LivenessProbe: &corev1.Probe{
+								ProbeHandler: corev1.ProbeHandler{
+									HTTPGet: &corev1.HTTPGetAction{
+										Path:   "/healthz",
+										Port:   intstr.FromInt(7979),
+										Scheme: corev1.URISchemeHTTP,
+									},
+								},
+								InitialDelaySeconds: 60,
+								PeriodSeconds:       60,
+								SuccessThreshold:    1,
+								FailureThreshold:    5,
+								TimeoutSeconds:      5,
+							},
+							Resources: corev1.ResourceRequirements{
+								Requests: corev1.ResourceList{
+									corev1.ResourceMemory: resource.MustParse("20Mi"),
+									corev1.ResourceCPU:    resource.MustParse("5m"),
+								},
+							},
+							VolumeMounts: []corev1.VolumeMount{
+								{
+									Name:      "credentials",
+									MountPath: "/etc/provider",
+								},
+							},
+						},
+					},
+					Volumes: []corev1.Volume{
+						{
+							Name: "credentials",
+							VolumeSource: corev1.VolumeSource{
+								Secret: &corev1.SecretVolumeSource{
+									SecretName: o.CredentialsSecret.Name,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	// Add platform specific settings
+	switch o.Provider {
+	case "aws":
+		deployment.Spec.Template.Spec.Containers[0].Env = append(deployment.Spec.Template.Spec.Containers[0].Env,
+			corev1.EnvVar{
+				Name:  "AWS_SHARED_CREDENTIALS_FILE",
+				Value: "/etc/provider/credentials",
+			},
+			corev1.EnvVar{
+				Name:  "AWS_REGION",
+				Value: "us-east-1",
```

Comment:
should `us-east-1` be dynamic input?

Reply:
Because external-dns only makes route53 calls, and route53 is global, we can hardcode this to us-east-1.  Otherwise, we would need to add `--external-dns-aws-region` as an additional flag on `hypershift install`